src/oauth2.cc

```diff
@@ -249,11 +263,14 @@ json::value OAuth2::ComputeTokenFromCredentials() const {
     return parsed_token;
   } catch (const json::Exception& e) {
     LOG(ERROR) << e.what();
-    return nullptr;
   } catch (const boost::system::system_error& e) {
     LOG(ERROR) << "HTTP error: " << e.what();
-    return nullptr;
   }
+
+  ::opencensus::stats::Record(
+    {{::google::GceApiRequestErrors(), 1}},
```

Comment:
This seems to be exposing very low-level details. Since we have to pre-register the measure and the tag anyway, would it not be better to have something like:
```c++
Metrics::RecordGceApiRequestErrors(1, "oauth2");
```
where `Metrics::RecordGceApiRequestErrors` is defined as:
```c++
class Metrics {
  void RecordGceApiRequestErrors(int64_t value, const std::string& method) {
    ::opencensus::stats::Record(
      {{gce_api_request_errors_, value}},
      {{method_tag_key_, method}});
  }
};
```
FWIW, I'm pretty sure there's an even better abstraction that would let us say something like:
```c++
Metrics::Record(
  "container.googleapis.com/internal/metadata_agent/gce_api_request_errors",
  {{"method", "oauth2"}});
```
and register these on-demand, where the value and tag types can be inferred from the template type arguments, but we can leave that for later.